src/ngraph/runtime/cpu/kernel/dropout.hpp

```diff
@@ -0,0 +1,85 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <random>
+
+#include "ngraph/shape.hpp"
+#include "ngraph/state/rng_state.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace cpu
+        {
+            namespace kernel
+            {
+                // Note: this kernel is for doing upscale in train
+                template <typename T>
+                void generate_dropout(T* input,
+                                      T* out0,
+                                      T* out1_mask,
+                                      size_t count,
+                                      bool training,
+                                      const double value)
+                {
+                    if (training)
+                    {
+                        double dropout_prob = 1 - value;
+                        size_t nthr = 28, nelems = count;
```

Comment:
Where is the magic number 28 coming from?

Reply:
fixed the magic number in new commit